### PR TITLE
Added AxisSpec file for Axis conversion tests

### PR DESCRIPTION
**Pull request by moneimne**
Fix for #5233.

### CODE REVIEW

File: Specs/Scene/AxisSpec.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        var upAxis = Cartesian4.UNIT_X;
+        var transformation = Axis.X_UP_TO_Z_UP;
+        var transformed = new Cartesian4();
+        Matrix4.multiplyByVector(transformation, upAxis, transformed);
+        var result = new Cartesian4();
+        Cartesian4.normalize(transformed, result);
+        expect(result).toEqualEpsilon(Cartesian4.UNIT_Z, CesiumMath.EPSILON1);
+    });
+
+}, 'WebGL');
```
Comment (lilleyse) [inline Specs/Scene/AxisSpec.js]: Remove `WebGL` since this test doesn't actually use it.

File: Specs/Scene/AxisSpec.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    'Core/Math',
+    'Core/Matrix3',
+    'Core/Matrix4'
+], function(
+    Axis,
+    Cartesian4,
+    CesiumMath,
+    Matrix3,
+    Matrix4) {
+    'use strict';
```
Comment (lilleyse) [inline Specs/Scene/AxisSpec.js]: Fix the indentation slightly: https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Specs/Core/sampleTerrainSpec.js#L2-L10

File: Specs/Scene/AxisSpec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    it('Convert x-up to z-up', function() {
+        var upAxis = Cartesian4.UNIT_X;
+        var transformation = Axis.X_UP_TO_Z_UP;
+        var transformed = new Cartesian4();
+        Matrix4.multiplyByVector(transformation, upAxis, transformed);
+        var result = new Cartesian4();
+        Cartesian4.normalize(transformed, result);
+        expect(result).toEqualEpsilon(Cartesian4.UNIT_Z, CesiumMath.EPSILON1);
+    });
```
Comment (lilleyse) [inline Specs/Scene/AxisSpec.js]: I bet this file could be shorted by creating a local function that takes three arguments, `upAxis`, `transformation`, and `expected`, which each of these tests call.

File: Specs/Scene/AxisSpec.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    it('Convert y-up to z-up', function() {
+        var upAxis = Cartesian4.UNIT_Y;
+        var transformation = Axis.Y_UP_TO_Z_UP;
+        var transformed = new Cartesian4();
+        Matrix4.multiplyByVector(transformation, upAxis, transformed);
+        var result = new Cartesian4();
+        Cartesian4.normalize(transformed, result);
+        expect(result).toEqualEpsilon(Cartesian4.UNIT_Z, CesiumMath.EPSILON1);
+    });
```
Comment (lilleyse) [inline Specs/Scene/AxisSpec.js]: Could be slightly more concise like:

```javascript
var upAxis = Cartesian4.UNIT_Y;
var transformation = Axis.Y_UP_TO_Z_UP;
var transformed = Matrix4.multiplyByVector(transformation, upAxis, new Cartesian4());
Cartesian4.normalize(transformed, transformed);
expect(transformed).toEqualEpsilon(Cartesian4.UNIT_Z, CesiumMath.EPSILON1);
```

File: Specs/Scene/AxisSpec.js (hunk 3 above)
Comment (moneimne) [inline Specs/Scene/AxisSpec.js]: Is it acceptable to put an `expect` call within the local function or is it better style/practice to return a value and test it in the `it` block?

Comment (lilleyse) [inline Specs/Scene/AxisSpec.js]: Definitely acceptable to put the `expect` in the local function.

**Comment by moneimne** [comment]
Thanks for the reviews, @lilleyse! The file has been updated.

**Comment by lilleyse** [comment]
Thanks @moneimne!

**Pull request merged**